sources/commands/option_external.c

```diff
@@ -6,20 +6,25 @@
 /*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/09/30 00:10:22 by inikulin          #+#    #+#             */
-/*   Updated: 2024/10/12 01:19:09 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/21 03:14:32 by inikulin         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "commands.h"
 
-static char	*find_executable(char *tgt, t_dlist *path)
+static char	*find_executable(char *tgt, t_dlist *path, int *errno)
 {
 	char	*fullpath;
 
 	while (path)
 	{
 		fullpath = ft_strjoin_multi_free_outer(
```

Comment:
3.1 take current node of param->envvar_path_head (a dlist node, a folder name), append the executable name to it
e.g. if the PATH is /home/ioann/shell:/usr/bin:/usr/sbin and the executable we're trying to call is ls, fullpath will try to be /home/ioann/shell/ls, /usr/bin/ls, /usr/sbin/ls